specs/dialog/dialog.md

```diff
@@ -0,0 +1,113 @@
+
+- Harold is in the process of removing old data sets from his company's content management system. He selects a group of data and clicks delete. A popup appears asking him to confirm that he wants to delete the selected files permanently.
+
+- Meredith is browsing news articles online and clicks a link to a local newspaper story about an upcoming heat wave in the region. She is informed that she has read through an allotted number of free articles for the month and must either login or become a subscriber of the news site to continue. Meredith tries to continue interacting with the web page, but she's unable to. Meredith decides to subscribe to the online newspaper because they are running a special.
+
+### Features
+
+- **Soft dismiss (light dismiss):** In certain implementations, app authors want a dialog to close when a user clicks outside the content region or presses the `esc` key with their keyboard.
+
+- **Focus management:** In all circumstances, focus should move to an element contained in the dialog. Per the W3C, "Unless a condition where doing otherwise is advisable, focus is initially set on the *first* focusable element" (emphasis mine). If an element within the dialog is set to be focused by an app author, our control should not override that behavior. In both modal and non-modal dialogs, tab focus should remain within the dialog content region and not move outside it. When a dialog the dialog is closed, users may also want a way to return focus to the element which invoked the dialog.
+
+### Prior Art/Examples
+- [FAST-DNA (React)](https://explore.fast.design/components/dialog)
+- [Material UI](https://material-ui.com/components/dialogs/)
+- [Lightning Design](https://www.lightningdesignsystem.com/components/modals/)
+- [Carbon Design](https://www.carbondesignsystem.com/components/modal/code/)
+- [Ant Design](https://ant.design/components/modal/)
+- [Atlassian](https://atlaskit.atlassian.com/packages/core/modal-dialog)
+- [Office Fabric](https://developer.microsoft.com/en-us/fabric#/controls/web/dialog)
+- [Windows (UWP)](https://docs.microsoft.com/en-us/windows/uwp/design/controls-and-patterns/dialogs-and-flyouts/dialogs)
+
+---
+
+## Design
+
+### API
+
+*Component name:*
+- `fast-dialog`
+
+*Attributes:*
+- `modal` - boolean value for modal functionality. Sets aria-modal to true.
+- `visible` - boolean value to toggle the visibility of the dialog
+- `aria-labelledby` - optional based on implementation**
+- `aria-describedby` - optional based on implementation**
+- `aria-label` - optional, based on implementation**
+
+** See the [W3C Specification](https://w3c.github.io/aria-practices/#dialog_roles_states_props) for requirements and details.
+
+*Custom properties:*
+- width
+- height
+
+*Slots:*
+- default slot for content
+
+*Events*
+- dismiss - event fired when a user presses escape or clicks outside the dialog content
```

Comment:
do want an event for opening also?

Reply:
Would we consider "opening" the same as visible in this case? I'm open to adding an additional event here, but am curious about the use case considering app authors would be managing the visibility and instantiation of the dialog fully.